### DISCUSSION TITLE
Loco Position System

### DISCUSSION THREAD

**Discussion by wangjinyi23** [bitcraze/discussions#1054] [Q&A]
Hello!We used to use the local position system to control crazyfile fixed-point flight, but the drone would fly randomly just after takeoff. What is the reason for this? I am not sure if it is necessary to use the ROS environment when using the local position system? Is there any relevant code for reference? Thank you very much!

**Comment by krichardsson** [comment]
Hi,
I converted your issue to a discussion instead.

As the Loco positioning system can not measure the yaw of the Crazyflie, it relies on that the Crazyflie is orientated in the positive x direction. After take off the estimator can keep track of the yaw by using the IMU and changes in the position. 

Also see the [tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-flying-using-lps/#restart-the-crazyflie)
